File: frontend/src/types/system.ts
export type HealthResponse = {
  status: string;
  service: string;
  environment: string;
  phase: string;
  data_backend?: string;
  database_configured_backend?: string;
  database_runtime_status?: string;
  database_runtime_message?: string;
  database_persistence_enabled?: boolean;
};

export type DatabaseSmokeReport = {
  configuredBackend: string;
  runtimeBackend: string;
  runtimeStatus: string;
  persistenceEnabled: boolean;
  runtimeMessage: string;
  productTotal: number;
  faqTotal: number;
  createdProductId: string;
  createdFaqEntryId: string;
  persistedRunId: string;
};

File: backend/app/schemas/system.py
from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: str
    service: str
    environment: str
    phase: str
    data_backend: str
    database_configured_backend: str
    database_runtime_status: str
    database_runtime_message: str
    database_persistence_enabled: bool


class DatabaseSmokeResponse(BaseModel):
    configured_backend: str
    runtime_backend: str
    runtime_status: str
    persistence_enabled: bool
    runtime_message: str
    product_total: int
    faq_total: int
    created_product_id: str
    created_faq_entry_id: str
    persisted_run_id: str
